server/store.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"encoding/json"
+)
+
+const (
+	timestampKey           = "timestamp"
+	userChannelActivityKey = "userChannelActivity"
+)
+
+type recommendedChannel struct {
+	ChannelID string  // identifier
+	Score     float64 // score
+}
+
+// initStore method is for initializing the KVStore.
+func (p *Plugin) initStore() error {
+	err := p.saveTimestamp(-1)
+	if err != nil {
+		return err
+	}
+	return p.saveUserChannelActivity(make(userChannelActivity))
+}
+
+// saveUserRecommendations saves user recommendations in the KVStore.
+func (p *Plugin) saveUserRecommendations(userID string, channels []*recommendedChannel) error {
+	return p.save(userID, channels)
+}
+
+// retreiveUserRecomendations gets user recommendations from the KVStore.
+func (p *Plugin) retreiveUserRecomendations(userID string) ([]*recommendedChannel, error) {
+	recommendations := make([]*recommendedChannel, 0)
+	err := p.retreive(userID, &recommendations)
+	return recommendations, err
+}
+
+// saveTimestamp saves timestamp in the KVStore.
+// All posts until this timestamp should already be analyzed.
+func (p *Plugin) saveTimestamp(time int64) error {
+	return p.save(timestampKey, time)
+}
+
+// retreiveTimestamp gets timestamp from KVStore.
+func (p *Plugin) retreiveTimestamp() (int64, error) {
+	var time int64
+	err := p.retreive(timestampKey, &time)
+	return time, err
+}
+
+// saveUserChannelActivity saves user-channel activity in the KVStore.
+func (p *Plugin) saveUserChannelActivity(activity userChannelActivity) error {
+	return p.save(userChannelActivityKey, activity)
+}
+
+// retreiveUserChannelActivity gets user-channel activity from the KVStore.
+func (p *Plugin) retreiveUserChannelActivity() (userChannelActivity, error) {
+	var act userChannelActivity
+	err := p.retreive(userChannelActivityKey, &act)
+	return act, err
+}
+
+// save method saves generic value in the KVStore
+func (p *Plugin) save(key string, value interface{}) (err error) {
+	j, err := json.Marshal(value)
```

Comment:
You may optionally use the newly added plugin helpers for persisting KV values and automatically marshalling. It's not technically broken, but I note that having a single `userChannelActivityKey` key means there could be a race as to whose value gets written when the plugin responds to requests asynchronously. I don't think there's a "wrong" answer to that, and the server won't corrupt the value, so this can stay, but just a heads up.